### PR TITLE
Shadowserver config fixups

### LINKED ISSUE #733: feed.* naming: clean up after PR #648
#648 added a "feed.name" to the config.py. However, as it turned out with a discussion between @dmth and me, we actually had two different concepts in mind:
- feeder (i.e. the source of the data feed) - let's call it now "feed.provider"
  and
- one or multiple feeds ("feed.name") that a "feed.provider" might provide.

We need to separate these concepts.

Action items:
- [x] revert back the addition of "feed.name" to config.py
- [ ] add a "feed.provider" to the DHO
- [ ] clarify DHO description
- [ ] add a "feed.provider" to the config.py  to the collector
- [ ] adapt test cases

### PR DESCRIPTION
**Pull request by aaronkaplan**
Partially fixes #733

### CODE REVIEW

**Comment by aaronkaplan** [comment]
merging now... will fix the xdmcp test case bug (build failing) very soon.

**Comment by sebix** [comment]
Beside the build fail, also see the pep8 output:

```
intelmq/bots/parsers/shadowserver/config.py:850:9: E265 block comment should start with '# '
intelmq/bots/parsers/shadowserver/config.py:881:9: E265 block comment should start with '# '
```

**Comment by sebix** [comment]
As there are so many merges and fixes in this PR, I'd prefer a rebase before merging (can also do it myself, if I can merge it)

**Comment by aaronkaplan** [comment]
> On 17 Oct 2016, at 17:31, Sebastian <notifications@github.com> wrote:
> 
> As there are so many merges and fixes in this PR, I'd prefer a rebase before merging (can also do it myself, if I can merge it)
> 
Feel free & thx,
a.

File: intelmq/bots/parsers/shadowserver/config.py [hunk 1]
```diff
@@ -889,9 +847,7 @@ def validate_ip(value):
         'classification.type': 'vulnerable service',
         'classification.taxonomy': 'Vulnerable',
         'protocol.application': 'xdmcp',
-        'feed.code': 'shadowserver-openxdmcp',
-        'feed.name': 'shadowserver',
-        'feed.url': 'https://www.shadowserver.org/wiki/pmwiki.php/Services/Open-XDMCP',
+        # 'feed.url': 'https://www.shadowserver.org/wiki/pmwiki.php/Services/Open-XDMCP',
```
Comment (sebix) [inline intelmq/bots/parsers/shadowserver/config.py]: This is not feed.url, it's the documentation, which is already linked above on line 825.

File: intelmq/bots/parsers/shadowserver/config.py [hunk 2]
```diff
@@ -922,9 +878,7 @@ def validate_ip(value):
         'classification.type': 'vulnerable service',
         'classification.taxonomy': 'Vulnerable',
         'protocol.application': 'nat-pmp',
-        'feed.code': 'shadowserver-opennatpmp',
-        'feed.name': 'shadowserver',
-        'feed.url': 'https://www.shadowserver.org/wiki/pmwiki.php/Services/Open-NATPMP',
+        # 'feed.url': 'https://www.shadowserver.org/wiki/pmwiki.php/Services/Open-NATPMP',
```
Comment (sebix) [inline intelmq/bots/parsers/shadowserver/config.py]: Same here, it's already on line 855.

**Comment by sebix** [comment]
@dmth could you review it too?

**Comment by aaronkaplan** [comment]
> On 18 Oct 2016, at 09:27, Sebastian <notifications@github.com> wrote:
> 
> @sebix requested changes on this pull request.
> 
> In intelmq/bots/parsers/shadowserver/config.py:
> 
> > @@ -889,9 +847,7 @@ def validate_ip(value):
>          'classification.type': 'vulnerable service',
>          'classification.taxonomy': 'Vulnerable',
>          'protocol.application': 'xdmcp',
> -        'feed.code': 'shadowserver-openxdmcp',
> -        'feed.name': 'shadowserver',
> -        'feed.url': 'https://www.shadowserver.org/wiki/pmwiki.php/Services/Open-XDMCP',
> +        # 'feed.url': 'https://www.shadowserver.org/wiki/pmwiki.php/Services/Open-XDMCP',
> 
> This is not feed.url, it's the documentation, which is already linked above on line 825.
> 

True. But are we strict with that in every bot?

**Comment by aaronkaplan** [comment]
> On 18 Oct 2016, at 09:33, L. Aaron Kaplan <aaron@lo-res.org> wrote:
> 
>> 
>> On 18 Oct 2016, at 09:27, Sebastian <notifications@github.com> wrote:
>> 
>> @sebix requested changes on this pull request.
>> 
>> In intelmq/bots/parsers/shadowserver/config.py:
>> 
>>> @@ -889,9 +847,7 @@ def validate_ip(value):
>>         'classification.type': 'vulnerable service',
>>         'classification.taxonomy': 'Vulnerable',
>>         'protocol.application': 'xdmcp',
>> -        'feed.code': 'shadowserver-openxdmcp',
>> -        'feed.name': 'shadowserver',
>> -        'feed.url': 'https://www.shadowserver.org/wiki/pmwiki.php/Services/Open-XDMCP',
>> +        # 'feed.url': 'https://www.shadowserver.org/wiki/pmwiki.php/Services/Open-XDMCP',
>> 
>> This is not feed.url, it's the documentation, which is already linked above on line 825.
>> 
> 
> True. But are we strict with that in every bot?
let me rephrase: is the feed.url for documentation purposes (pointing the recipient to a URL for the feed) or for automation purposes (so that the recipient can pull down the feed him/herself)?
In the second case, I agree . In the first case I thing it's okay to add the documentation URL.

So far I was under the impression that our intention was the first approach.
However, then the name "feed.url" might be wrong.

**Comment by dmth** [comment]
Concerning the correct understanding of `feed.url`: We should agree on one interpretation and document this interpretation in our DHO.

**Comment by aaronkaplan** [comment]
> On 18 Oct 2016, at 09:52, Dustin Demuth <notifications@github.com> wrote:
> 
> Concerning the correct understanding of feed.url: We should agree on one interpretation and document this interpretation in our DHO.
> 
Exactly!

**Comment by sebix** [comment]
The docs only say `The URL of a given abuse feed, where applicable` for `feed.url`.

**Comment by sykaeh** [comment]
I understood `feed.url` to be the source of the actual feed, not the documentation of the feed, but I do see the value of providing a URL that documents the feed. Maybe add a new field `feed.documentation`?

**Comment by aaronkaplan** [comment]
> On 18 Oct 2016, at 11:04, Sybil Ehrensberger <notifications@github.com> wrote:
> 
> I understood feed.url to be the source of the actual feed, not the documentation of the feed, but I do see the value of providing a URL that documents the feed. Maybe add a new field feed.documentation?
> 

Yeah... recently we are stumbling across a lot of unclear language in the DHO due to simplifications which we made in the beginning.
On the other hand - creating a field for every possible interpretation might be too much as well.

Tough call. But good that we understood that the DHO was not clear here.

**Comment by sebix** [comment]
A rebased PR is in branch [dev-shadowserver-fixups](https://github.com/certtools/intelmq/tree/dev-shadowserver-fixups)

**Comment by dmth** [comment]
I split of the discussion about the feed.url into https://github.com/certtools/intelmq/issues/749

**Comment by dmth** [comment]
@sebix as a rebased branch containing the changes in this PR exist, does also a new PR for this branch exist which is easiser to review? If yes, this PR might be closed for the sake of clarity.

**Pull request closed without merging**